designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentService.scala

```diff
@@ -31,6 +32,12 @@ trait ProcessStateService {
   def getProcessState(processIdWithName: ProcessIdWithName)
                      (implicit user: LoggedUser, ec: ExecutionContext): Future[ProcessState]
 
+  // This method in contrary to getProcessState doesn't invoke target DeploymentManager - it only compute state
+  // based on information available in DB
+  // TODO: add caching of state returned by DeploymentManager
+  def getDbProcessState(processDetails: BaseProcessDetails[_])
```

Comment:
Why there is Db in the method name? I think it's just technical details?

Reply:
I was looking for some better word but I couldn't find it. Maybe you have some better candidate? Currently, we have:
1. "real state" based on remote state on engine + local 
2. only local state

local state - not local for designer, but local in our database. I wanted to distinguish between them so I call the second one "Db" + add this comment. So in fact, all we do here is about technical, performance details :)

 I want to introduce caching in next PR, so this name probably will be changed.

Reply:
`getInitialProcessState` - as I understood, it is state before we call the manager?

Reply:
nope it is totally different state, it is not "initial" state. "initial" is rather "not deployed". It is local db state, without knowledge from remote engine (without calling DM)

Reply:
`getInternalProcessState`? But i am not sure also.. For me `getDbProcessState` might stay

Reply:
I prefer Internal rather than DB... But still, we compute here the initial state from our storage (in this case it's the DB).